Fix: Preserve local and custom model names in parse_model_name

ps2program opened the pull request:
## Summary

This PR improves the `parse_model_name` utility in `deepeval.models.utils` by ensuring that model names with non-standard or custom prefixes like `local/llama-3` are preserved and not truncated.

### Problem

The current implementation of `parse_model_name` assumes that any string before a `/` is a provider name, and always strips it. This causes issues for locally hosted or custom models like `local/llama-3`, which get incorrectly truncated to just `llama-3`.

### Fix

- Added logic to preserve the full model name if the prefix is not in a known list of providers (e.g., `"openai"`, `"anthropic"`, `"cohere"`).
- Updated docstring for clarity.
- Updated tests to:
  - Expect `None` for `None` input instead of raising `TypeError`.
  - Add new test case to verify local/custom model names are preserved (`local/llama-3`, `custom/model-xyz`).

### Example Behavior

```python
parse_model_name("openai/gpt-4o")       # "gpt-4o"
parse_model_name("local/llama-3")       # "local/llama-3"
parse_model_name("gpt-4o")              # "gpt-4o"
parse_model_name(None)                  # None

penguine-ip commented:
hey @ps2program actually we're thinking about removing this functino entirely. The reason why this was introduced is because we wanted to fit the litellm format of "provider/model" but i'd better for us to just support lite llm directly. Was wondering can you do that PR instead?

ps2program commented:
**Updated PR:**
Removed `parse_model_name` function and updated usages and tests to support full `provider/model` format

**Description:**

* Removed the `parse_model_name` utility function, as per recent decision to fully support LiteLLM’s `provider/model` naming format directly without parsing or stripping prefixes.
* Updated all code references to use model names in full format without modifications.
* Modified and removed related tests for `parse_model_name`. Added new test suite to ensure model names are accepted as-is in both `provider/model` and simple model name formats.
* Cleaned up usages in `base_model.py`, `azure_model.py`, `anthropic_model.py`, `openai_model.py`, and their corresponding tests.
* Confirmed that the system now treats model names as full strings, consistent with LiteLLM integration expectations.

Please review and let me know if there are any edge cases or legacy concerns to address.
